p2p/pkg/notifications/notifications.go

```diff
@@ -0,0 +1,92 @@
+package notifications
```

Comment:
I'd suggest to rename the package to `notification`.

Reply:
Any specific reason? Usually I name package name as collection rather than single.